src/views/AddStation.js

```diff
@@ -0,0 +1,186 @@
+import React, { useState, Fragment } from "react";
+import axios from "axios";
+import { API_ROOT, toolTipCopy } from "../app/app-helpers";
+import { useAuthState } from "../auth/auth-context";
+import { QuestionMarkToolTip } from "../app/app-helpers";
+import { checkJwt } from "../auth/auth-helpers";
+import Spinner from "../app/components/Spinner";
+import CircleCheck from "../assets/CircleCheck.svg";
+
+export default function AddStation() {
+  const { jwt } = useAuthState();
+  // form state
+  const [stationName, setStationName] = useState(``);
+  const [latitude, setLatitude] = useState(``);
+  const [longitude, setlongitude] = useState(``);
+  const [altitude, setAltitude] = useState(``);
+  const [notes, setNotes] = useState(``);
+  // submission state
+  const [isLoading, setIsLoading] = useState(false);
+  const [successfullyAddedStation, setSuccessfullyAddedStation] = useState(``);
+  const [errors, setErrors] = useState([]);
+  const [isError, setIsError] = useState(false);
+
+  const resetFormValues = () => {
+    setStationName(``);
+    setLatitude(``);
+    setlongitude(``);
+    setAltitude(``);
+    setNotes(``);
+  };
+
+  const submitLocation = async () => {
+    setIsError(false);
+    setSuccessfullyAddedStation(``);
+    setIsLoading(true);
+    // checks if jwt is valid and hasn't expired
+    checkJwt(jwt);
+    // if no errors
+    try {
+      const result = await axios.post(
+        `${API_ROOT}/generateStation`,
+        JSON.stringify({
+          jwt: jwt,
+          station: stationName,
+          latitude: latitude,
+          longitude: longitude,
+          altitude: altitude,
+          notes: notes
+        })
+      );
+      console.log(result);
+      setSuccessfullyAddedStation(result.data.station_id);
+    } catch (error) {
+      console.log(error);
+      setIsError(true);
+    }
+    setIsLoading(false);
+    resetFormValues();
+  };
+
+  return (
+    <div className="add-station">
+      <h1 className="static-page__main-header--small">Add a location</h1>
+      <h2 className="add-station__sub-header">
+        You first need to set a station location
+      </h2>
+
+      <form
+        className="app__form"
+        onSubmit={event => {
+          event.preventDefault();
+          submitLocation();
+        }}
+      >
+        <div>
+          <label className="app__form__label station-form__label">
+            <p>Station name or label</p>
+            <QuestionMarkToolTip toolTipText={toolTipCopy.station_name} />
+          </label>
+          <input
+            required
+            type="text"
+            className="app__form__input"
+            value={stationName}
+            onChange={event => setStationName(event.target.value)}
+            maxLength={30}
+            placeholder="e.g. Crater Lake Campground"
+          ></input>
+        </div>
+        <div>
+          <label className="app__form__label station-form__label">
+            <p>Latitude (degrees)</p>
```

Comment:
Add a warning that lat, long and alt are permanent and can't be edited. The observations you make will be associated with this location and it cannot be changed. If you have a need for another location, just add another.